### DISCUSSION TITLE
PGO on cargo release profile?

### DISCUSSION THREAD

**Discussion by rust-academy** [Kobzol/cargo-pgo#22] [Q&A] [answered]
Hi, 
great project, thanks for the work. 

I ran cargo-pgo on a relatively simple actix application and saw some good performance improvements. 

However, the pgo optimization is a lot better than the cargo dev profile, but not quite as good as the release profile with LTO & strip. I use the following  release config: 

````toml
[profile.release]
codegen-units = 1  # default is 16 
lto = true         # default is false 
strip = true       # reduces binary size
panic = "abort" 
````

Adding release to the build command resulted in an error. 

````bash
cargo pgo instrument build release 
````

The question I have is, how do I use PGO on the release profile? 
Any thoughts? 

-
Marvin

**Comment by Kobzol** [comment]
Hi, `cargo pgo` internally adds `--release` to all builds commands, so everything should be built in release mode. It should also respect the `[profile.release]` configuration from your `Cargo.toml` file.

**Comment by rust-academy** [comment]
Thank you for the clarification.
Will measure again tomorrow.

Would you please add this to the documentation?

**Comment by Kobzol** [comment, marked as answer]
It's written in the README (https://github.com/Kobzol/cargo-pgo#pgo). Is there another place where you would like to see this information?

`cargo-pgo` also warns about this, but you have to execute the command with the proper flag (`--release` vs `release`):
```bash
$ cargo pgo build -- --release
$ [WARN] Do not pass `--release` manually, it will be added automatically by `cargo-pgo`
```

**Comment by rust-academy** [comment]
Yes, you're right of course.  
I'll take the liberty of closing this.
